### ISSUE TITLE
Refactor and generalize PyTorch collate function to preserve sample sequences and leverage default_collate

### ISSUE THREAD

**Issue by Yann-CV** [goldener-data/goldener#254]
**Context & Problem**

In Goldener, intermediate and final results are stored in Pixeltable Tables, which can be converted into PyTorch Datasets. Currently, primitive types and arrays from these datasets may be returned as native Python types (int, np.ndarray) rather than torch.Tensor.

To handle batching, we previously introduced pxt_torch_dataset_collate_fn. However, we have identified two key issues:

- Over-specific design: We initially assumed this collate behavior was only needed for datasets coming directly from Pixeltable. In reality, it is useful for any PyTorch Dataset that formats sample items as a dictionary ready for Goldener tools.

- Standard PyTorch collate_fn list splitting: PyTorch's native collation treats items inside a list as individual features to collate across samples. For example:

Input: [{"label": [class_1, class_2]}, {"label": [class_1, class_2]}]
Standard PyTorch output: batch["label"] = [('class_1', 'class_1'), ('class_2', 'class_2')]
Expected Goldener output: batch["label"] = [('class_1', 'class_2'), ('class_1', 'class_2')]

Additionally, PyTorch raises an error if list lengths vary across samples because it attempts element-wise collation. In Goldener, we operate under the strong assumption that each dataset sample is a dict, and each key-value pair in that dict represents a single complete feature. Sequences should be preserved per sample, leaving tensor conversion to downstream tools when necessary.

**Proposed Solution**

Move the collate function to goldener/utils/torch_utils.py (or the corresponding utility module) and rename it to collate_keeping_sequences_as_sequences to reflect its general utility.

Refactor the logic to delegate non-sequence feature collation directly to PyTorch's standard torch.utils.data.default_collate.
 
**Suggested Implementation**

```
def collate_keeping_sequences_as_sequences(batch: list[dict[str, Any]])  -> dict[str, torch.Tensor | Sequence]:
    value_list_dict = defaultdict(list)
    conversion_dict: dict[str, Callable[[Sequence[Any]], torch.Tensor]] = {}

    for idx_sample, sample in enumerate(batch):
        for key, value in sample.items():
            value_list_dict[key].append(value)
            if idx_sample == 0:
                if not isinstance(value, Sequence):
                    conversion_dict[key] = default_collate

    return {
        key: conversion_dict[key](value) if key in conversion_dict else value
        for key, value in value_list_dict.items()
    }
```

**Comment by Yann-CV** [comment]
This issue is linked to a known bug: https://github.com/goldener-data/goldener/issues/202
